internal/server/stat/stat_util.go

```diff
@@ -248,3 +249,35 @@ func getMetadataHash(m *pb.StatMetadata) uint32 {
 	}, "-")))
 	return h.Sum32()
 }
+
+// getSourceSeriesKey computes the key for *pb.SourceSeries.
+// The key is based on the unique properties of the observations.
+func getSourceSeriesKey(s *pb.SourceSeries) string {
+	return fmt.Sprintf(
+		"%s_%s_%s_%s_%s",
+		s.ImportName,
+		s.MeasurementMethod,
+		s.ObservationPeriod,
+		s.ScalingFactor,
+		s.Unit)
+}
+
+// MergeSourceSeries merges lists of SourceSeries. For same source series, the one
+// with the latest data will be used.
```

Comment:
MergeSourceSeries gives the sense that a single list gets merged... perhaps worth noting that never happens.

Wonder if the function should be named like CollectDistinctSourceSeries(), or similar...

Reply:
Done